src/System.Net.Http/tests/StressTests/HttpStress/ClientOperations.cs

```diff
@@ -149,21 +179,55 @@ public static class ClientOperations
                     using (var req = new HttpRequestMessage(HttpMethod.Get, "/headers") { Version = httpVersion })
                     using (HttpResponseMessage m = await ctx.SendAsync(req))
                     {
-                        ValidateResponse(m, httpVersion);
+                        ValidateHttpVersion(m, httpVersion);
+                        ValidateStatusCode(m);
                         ValidateContent(ctx.ContentSource, await m.Content.ReadAsStringAsync());
                     }
                 }),
 
+                ("GET Echo Headers",
+                async ctx =>
+                {
+                    Version httpVersion = ctx.GetRandomHttpVersion();
+
+                    using (var req = new HttpRequestMessage(HttpMethod.Get, "/echoHeaders") { Version = httpVersion })
+                    {
+                        ctx.PopulateWithRandomHeaders(req.Headers);
+
+                        using (HttpResponseMessage res = await ctx.SendAsync(req))
+                        {
+                            ValidateHttpVersion(res, httpVersion);
+                            ValidateStatusCode(res);
+
+                            // Validate that response headers are being echoed
+                            foreach (var reqHeader in req.Headers)
+                            {
+                                if (!res.Headers.TryGetValues(reqHeader.Key, out var values))
+                                {
+                                    throw new Exception($"Expected response header name {reqHeader.Key} missing.");
+                                }
+                                else if (!reqHeader.Value.SequenceEqual(values))
+                                {
+                                    string FmtValues(IEnumerable<string> values) => $"{string.Join(", ", values.Select(x => $"\"{x}\""))}";
```

Comment:
It looks like string interpolation isn't adding anything here.

Reply:
Saves a couple of characters and one intermediate allocation compared to having two `+`. Probably insignificant in this context though.

Reply:
I just realised you meant the outer string 🤦‍♂ . Fixing.